examples/edge_bundling.ipynb

```diff
@@ -0,0 +1,524 @@
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "render_points(nodes_df, width=2000, height=2000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Graph without edge bundling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "nop_df = nop_bundle(nodes_df, edges_df)\n",
+    "render_lines(nop_df, width=2000, height=2000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Graph with edge bundling"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%time df = bundle(nodes_df, edges_df, 0.05, 0.7)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "render_lines(df, width=2000, height=2000, cmap='blue')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "render_lines(df, width=2000, height=2000, cmap='red')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "render_lines(df, width=2000, height=2000, cmap=inferno, bgcolor='black')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "render_lines(df, width=2000, height=2000, cmap=viridis, bgcolor='black')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Edge bundling with random graph"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def generate_nodes(n):\n",
+    "    return pd.DataFrame(np.random.randn(n, 2), columns=['x', 'y'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def generate_edges(n, nodes):\n",
+    "    return pd.DataFrame(np.random.randint(len(nodes), size=(n, 2)), columns=['source', 'target'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def generate_random_graph(nodes, edges):\n",
+    "    ndf = generate_nodes(nodes)\n",
+    "    edf = generate_edges(edges, ndf)\n",
+    "    return ndf, edf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "nodes_df, edges_df = generate_random_graph(10000, 50000)"
+   ]
```

Comment:
It's hard to work with this notebook when the variables are redefined -- revisiting earlier cells to make a change suddenly uses a different graph.  Can you make sure that all distinct graphs are given distinct names?  (researchers_nodes, researchers_edges, circular_nodes, etc.)